src/cpu/flexcpu/simple_dataflow_cpu.cc

```diff
@@ -249,6 +249,45 @@ SimpleDataflowCPU::requestInstruction(const RequestPtr& req,
     return true;
 }
 
+bool
+SimpleDataflowCPU::requestMemRead(const RequestPtr& req, ThreadContext* tc,
+                                  MemCallback callback_func)
+{
+    // TODO may need to enqueue request if port is busy
```

Comment:
Make sure there isn't an infinite buffer here. For all of the places where there could be queuing make sure that we don't have implicit infinite queues. When testing, you should test with queue sizes of 1 to bring out bugs.